### ISSUE TITLE
ArrayShape does not support `list()`.

### ISSUE THREAD

**Issue by limingxinleo** [JetBrains/phpstorm-attributes#15]
```
<?php

#[\JetBrains\PhpStorm\ArrayShape(['int', \Throwable::class])]
function foo()
{
    return [];
}

$ret = foo();
$ret[1]->getMessage(); // Ok

[$id, $driver] = foo();
$driver->getMessage(); // Not work

```

**Comment by hailwood** [comment]
https://youtrack.jetbrains.com/issue/WI-60743

And from the issue that's marked as duplicate of it - 
> This problem is now fixed and the update to the 2021.3 EAP build will be available at the end of September.

**Comment by wbars** [comment]
The following code example is working in the latest EAP version.
